corehq/apps/userreports/reports/util.py

```diff
@@ -36,19 +39,38 @@ class ReportExport(object):
     """Export all the rows of a UCR report
     """
 
-    def __init__(self, domain, title, report_config, lang, filter_values):
+    def __init__(self, domain, title, report_config, lang, filter_values, request_user=None):
         self.domain = domain
         self.title = title
         self.report_config = report_config
         self.lang = lang
         self.filter_values = filter_values
+        self.request_user = request_user
 
     @property
-    @memoized
     def data_source(self):
         from corehq.apps.userreports.reports.data_source import ConfigurableReportDataSource
         data_source = ConfigurableReportDataSource.from_spec(self.report_config, include_prefilters=True)
         data_source.lang = self.lang
+        # Removing location from the filters for the locations that are not applicable for the current user.
+        if (toggles.LOCATION_RESTRICTED_SCHEDULED_REPORTS.enabled(self.domain)
```

Comment:
Doesn't this better fit in `ConfigurableReportView.filter_values`, that way we don't even need to pass `request_user` to [here](https://github.com/dimagi/commcare-hq/pull/32964/files#diff-25ce12893ad4531468d0ef019927e6cebbf9f5b4189a2c8d4e23981240b25784R42)?

Reply:
Agree. Changed accordingly.
https://github.com/dimagi/commcare-hq/compare/4aa8f3e56bf1b2a71519ce86107f8f83de12182f..2d86da47398b5f6870103c86809f590a0525bbbd